src/java/org/apache/cassandra/locator/RangesAtEndpoint.java

```diff
@@ -302,6 +303,11 @@ public static RangesAtEndpoint toDummyList(Collection<Range<Token>> ranges)
                 .collect(collector(dummy));
     }
 
+    public static boolean isDummyList(RangesAtEndpoint ranges)
```

Comment:
How does all replicas being local mean it's a dummy list? Only the 2nd half of this predicate is dummy list related.